hs_core/tasks.py

```diff
@@ -443,6 +443,61 @@ def copy_resource_task(ori_res_id, new_res_id=None, request_username=None):
         raise utils.ResourceCopyException(str(ex))
 
 
+@shared_task
+def replicate_resource_bag_to_user_zone_task(res_id, request_username):
+    """
+    Task for replicating resource bag which will be created on demand if not existent already to iRODS user zone
+    Args:
+        res_id: the resource id with its bag to be replicated to iRODS user zone
+        request_username: the requesting user's username to whose user zone space the bag is copied to
+
+    Returns:
+    None, but exceptions will be raised if there is an issue with iRODS operation
+    """
+
+    res = utils.get_resource_by_shortkey(res_id)
+    res_coll = res.root_path
+    istorage = res.get_irods_storage()
+    bag_modified_flag = True
+    # needs to check whether res_id collection exists before getting/setting AVU on it to
+    # accommodate the case where the very same resource gets deleted by another request when
+    # it is getting downloaded
+    if istorage.exists(res_coll):
+        bag_modified = istorage.getAVU(res_coll, 'bag_modified')
+
+        # make sure bag_modified_flag is set to False only if bag exists and bag_modified AVU
+        # is False; otherwise, bag_modified_flag will take the default True value so that the
+        # bag will be created or recreated
+        if bag_modified:
```

Comment:
Is this necessary?  I don't think we do this anywhere else we check the bag_modified avu.

Reply:
Yes, we do this in bag download as well as shown here: https://github.com/hydroshare/hydroshare/blob/develop/django_irods/views.py#L209-L215. It'd be good to move it to a utility method to remove code duplication, though. I'll take care of this in this PR then.